File: avalon-support/avalon-oauth2-support/avalon-oauth2-authorization-server/src/test/java/io/github/lishangbu/avalon/oauth2/authorizationserver/autoconfiguration/AuthorizationServerSettingsAutoConfigurationTest.java
package io.github.lishangbu.avalon.oauth2.authorizationserver.autoconfiguration;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

import io.github.lishangbu.avalon.oauth2.common.properties.Oauth2Properties;
import org.junit.jupiter.api.Test;
import org.springframework.security.oauth2.server.authorization.settings.AuthorizationServerSettings;

class AuthorizationServerSettingsAutoConfigurationTest {

    @Test
    void usesIssuerWhenAbsolute() {
        AuthorizationServerSettingsAutoConfiguration configuration =
                new AuthorizationServerSettingsAutoConfiguration();
        Oauth2Properties properties = new Oauth2Properties();
        properties.setIssuerUrl("https://issuer.example");

        AuthorizationServerSettings settings =
                configuration.authorizationServerSettings(properties);

        assertEquals("https://issuer.example", settings.getIssuer());
    }

    @Test
    void ignoresNonAbsoluteIssuer() {
        AuthorizationServerSettingsAutoConfiguration configuration =
                new AuthorizationServerSettingsAutoConfiguration();
        Oauth2Properties properties = new Oauth2Properties();
        properties.setIssuerUrl("/issuer");

        AuthorizationServerSettings settings =
                configuration.authorizationServerSettings(properties);

        assertNull(settings.getIssuer());
    }
}

File: avalon-platform/avalon-security/avalon-oauth2-authorization-server/src/test/kotlin/io/github/lishangbu/avalon/oauth2/authorizationserver/autoconfiguration/AuthorizationServerSettingsAutoConfigurationTest.kt
package io.github.lishangbu.avalon.oauth2.authorizationserver.autoconfiguration

import io.github.lishangbu.avalon.oauth2.common.properties.Oauth2Properties
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test

class AuthorizationServerSettingsAutoConfigurationTest {
    @Test
    fun usesIssuerWhenAbsolute() {
        val configuration = AuthorizationServerSettingsAutoConfiguration()
        val properties = Oauth2Properties()
        properties.issuerUrl = "https://issuer.example"

        val settings = configuration.authorizationServerSettings(properties)

        assertEquals("https://issuer.example", settings.issuer)
    }

    @Test
    fun ignoresNonAbsoluteIssuer() {
        val configuration = AuthorizationServerSettingsAutoConfiguration()
        val properties = Oauth2Properties()
        properties.issuerUrl = "/issuer"

        val settings = configuration.authorizationServerSettings(properties)

        assertNull(settings.issuer)
    }
}
